.github/workflows/build-test.yml

```diff
@@ -81,13 +110,27 @@ jobs:
                 cloud_provider: [ 'AWS', 'AZURE', 'GCP' ]
         steps:
             - uses: actions/checkout@v1
+            - name: Restore cached deps
+              id: cache-restore-deps
+              uses: actions/cache/restore@v3
+              with:
+                path: dep-cache
+                key: ${{ matrix.build_type }}-${{ matrix.platform }}-${{ matrix.vs_version }}-${{ github.event.pull_request.base.sha }}-Win-dep-cache
+              if: github.event_name == 'pull_request'
```

Comment:
should we always restore or should we check explicitly if the entry is available?